specification/cluster/_types/ComponentTemplate.ts

```diff
@@ -18,14 +18,11 @@
  */
 
 import { AliasDefinition } from '@indices/_types/AliasDefinition'
+import { DataStreamLifecycleWithRollover } from '@indices/_types/DataStreamLifecycle'
 import { IndexSettings } from '@indices/_types/IndexSettings'
 import { Dictionary } from '@spec_utils/Dictionary'
 import { IndexName, Metadata, Name, VersionNumber } from '@_types/common'
 import { TypeMapping } from '@_types/mapping/TypeMapping'
-import {
-  DataStreamLifecycle,
```

Comment:
And this is the import that actively misled me, as it shows up when I asked my editor to find references or `DataStreamLifecycle`.